package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+#     pylint: disable=E1101
+from __future__ import division
+
+import os.path as path
+import numpy as np
+
+from MDAnalysis.analysis.base import AnalysisBase
+
+class LinearDensity(AnalysisBase):
+    def __init__(self, universe, selection, grouping='atoms', binsize=0.25,
+                 start=None, stop=None, step=None):
+        self._ags = [selection] # allows use of run(parallel=True)
+        self._setup_frames(universe, start, stop, step)
+
+        self.binsize = binsize
+
+        # group of atoms on which to compute the COM (same as used in
+        # AtomGroup.wrap())
+        self.grouping = grouping
+
+        # Dictionary containing results
+        self.results = {'x': {'dim': 0}, 'y': {'dim': 1}, 'z': {'dim': 2}}
+        # Box sides
+        self.dimensions = [side for side in self._universe.dimensions[0:3]]
+        self.volume = np.prod(self.dimensions)
+        bins = [int(side//self.binsize) for side in self.dimensions] # number of bins
+
+        # Here we choose a number of bins of the largest cell side so that
+        # x, y and z values can use the same "coord" column in the output file
+        self.nbins = max(bins)
+        slices_vol = [self.volume/n for n in bins]
+
+        self.keys = ['pos', 'pos_std', 'char', 'char_std']
+
+        # Initialize results array with zeros
+        for dim in self.results:
+            idx = self.results[dim]['dim']
+            self.results[dim].update({'slice volume': slices_vol[idx]})
+            for key in self.keys:
+                self.results[dim].update({key: np.zeros(self.nbins)})
+
+        # Variables later defined in _prepare() method
+        self.masses = None
+        self.charges = None
+        self.totalmass = None
+
+    def _prepare(self):
+        # group must be a local variable, otherwise there will be
+        # issues with parallelization
+        group = getattr(self._ags[0], self.grouping)
+
+        # Get masses and charges for the selection
+        try: # in case it's not an atom
+            self.masses = np.array([elem.total_mass() for elem in group])
+            self.charges = np.array([elem.total_charge() for elem in group])
+        except AttributeError: # much much faster for atoms
+            self.masses = np.array(self._ags[0].masses)
```

Comment:
You don't need to call `np.array` on these because they're always returned as arrays anyway